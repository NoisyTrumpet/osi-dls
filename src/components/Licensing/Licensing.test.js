import React from 'react';
import { render, screen } from '@testing-library/react';
import { toBeInTheDocument } from '@testing-library/jest-dom/matchers';
import '@testing-library/jest-dom/extend-expect';
import { Licensing } from './';

expect.extend({
  toBeInTheDocument,
});

const BvSLogo =
  '../../conf/storybook/images/batman-vs-superman-color_pdp@1x.png';
const DCLogo = '../../conf/storybook/images/dc-comics_pdp@1x.png';

const logos = [
  {
    url: BvSLogo,
    alt: 'Batman v Superman',
  },
  {
    url: DCLogo,
    alt: 'DC Comics',
  },
];

const textToRender = 'BATMAN V SUPERMAN: DAWN OF JUSTICE';

describe('<Licensing />', () => {
  it('Should render', () => {
    const { container } = render(
      <Licensing logos={logos} text={textToRender} />,
    );
    screen.debug();
    expect(container.firstChild).toBeInTheDocument();
    expect(screen.getByText(textToRender)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
