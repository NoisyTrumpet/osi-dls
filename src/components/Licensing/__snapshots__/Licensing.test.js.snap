// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Licensing /> Should render 1`] = `
<div>
  <div
    class="licensing"
  >
    <div
      class="licensing-logos"
    >
      <div
        class="LazyLoad lazy-image-container"
      />
      <div
        class="LazyLoad lazy-image-container"
      />
    </div>
    <span
      class="typography typography-licensing"
    >
      BATMAN V SUPERMAN: DAWN OF JUSTICE
    </span>
  </div>
</div>
`;
